### PR TITLE
EIP-1559 support

### LINKED ISSUE #4327: Support EIP-1559 changes
In order to support EIP 1559 in Blockscout for xDai chain and Sokol, the following changes should be made. 
image

**Blocks**
Include the following data to the block pane:
- Burnt Fees
   Tooltip: Amount of xDai burned from transactions in this block (Base Fee * Gas Used in Block).
   
- Priority Fees
  Tooltip: Additional fees (tips) paid to the block validator for including transactions in this block. 

**Block Details**
- Base Fee per Gas
   Tooltip: Minimum fee required per unit of gas. Fee adjusts based on network congestion.
   
- Burnt Fees 
   Tooltip: xDai burned from transactions included in the block (Base fee (per unit of gas) * Gas Used).
   
- Priority Fee / Tip
 Tooltip: User-defined tips sent to validator for transaction priority/inclusion.
 
- Gas Target 
 Tooltip: Gas target regulates the Base Fee. Blocks filled above the Gas Target increase the Base Fee by 12.5% for the next block, blocks with gas usage below the target decrease the Base Fee by 12.5%. 

**Transaction details**
- Transaction Burned Fee 
   Tooltip: Amount of xDai burned for this transaction. Equals Block Base Fee per Gas * Gas Used.

- Priority Fee / Tip
  Tooltip: User-defined tip sent to validator for transaction priority/inclusion.

- Max Fee per Gas
  Tooltip: Maximum total amount per unit of gas a user is willing to pay for a transaction, including base fee and priority fee.

- Max Priority Fee per Gas
  Tooltip: User defined maximum fee (tip) per unit of gas paid to validator for transaction prioritization.
  
- Transaction Type 
   Tooltip: New transaction type identifier introduced in EIP 2718 (Berlin HF)
   
**Chain data**
  - Total Fees Burned
     Tooltip: Total Sum of Fees Burnt since the introduction of EIP 1559
     
**Dune Analytics**

- Total fees burned
- Total priority fees
- Total fees burned daily (last 7 and 30 days)
- Bar chart that shows Base fees + Priority fees stacked, daily (last 7 and 30 days)

### LINKED ISSUE #4490: Add index position of transaction in the block
Screenshot 2021-08-06 at 18 08 47

Add it together to the nonce field

### PR DESCRIPTION
**Pull request by nikitosing**
Close #4327 
## Changelog
- new env variable `ENABLE_1559_SUPPORT`
- new fields in DB
- support eip-1559 on transaction's page
- support eip-1559 on block's page
- Burnt Fees and Priority Fees for blocks are calculating at the runtime and storing in ets

### CODE REVIEW

File: apps/explorer/lib/explorer/chain.ex [hunk 1]
```diff
@@ -3898,7 +3898,7 @@ defmodule Explorer.Chain do
 
   defp fetch_transactions(paging_options \\ nil) do
     Transaction
-    |> order_by([transaction], desc: transaction.block_number, desc: transaction.index)
+    |> order_by([transaction], desc: transaction.block_number, asc: transaction.index)
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain.ex:3901]: @nikitosing  Let's move this change to a separate PR

File: apps/block_scout_web/lib/block_scout_web/templates/transaction/overview.html.eex [hunk 2]
```diff
@@ -314,26 +320,76 @@
...
+              <dd class="col-sm-9 col-lg-10"> <%= type %> </dd>
+            </dl>
+          <% end %>
           <hr>
           <!-- Gas Limit -->
           <dl class="row">
             <dt class="col-sm-3 col-lg-2 text-muted">
               <%= render BlockScoutWeb.CommonComponentsView, "_i_tooltip_2.html",
-                text: gettext("Maximum amount of gas provided for the transaction. For normal Eth transfers, the value is 21,000. For contract this value is higher and bound by block gas limit.") %>
+                text: gettext("Maximum gas amount approved for the transaction.") %>
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/templates/transaction/overview.html.eex:343]: @nikitosing let's move update of tooltips' text to a separate PR

File: apps/block_scout_web/lib/block_scout_web/templates/transaction/overview.html.eex [hunk 3]
```diff
@@ -343,16 +399,16 @@
           <dl class="row">
             <dt class="col-sm-3 col-lg-2 text-muted">
               <%= render BlockScoutWeb.CommonComponentsView, "_i_tooltip_2.html",
-                text: gettext("Sequential running number for an address, beginning with 0 for the first transaction. For example, if the nonce of a transaction is 10, it would be the 11th transaction sent from the sender's address.") %>
-              <%= gettext "Nonce" %>
+                text: gettext("Transaction number from the sending address. Each transaction sent from an address increments the nonce by 1.") %>
+              <%= gettext "Nonce | Index in block" %>
             </dt>
-            <dd class="col-sm-9 col-lg-10"> <%= @transaction.nonce %> </dd>
+            <dd class="col-sm-9 col-lg-10"> <%= @transaction.nonce %> | <%= @transaction.index %> </dd>
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/templates/transaction/overview.html.eex:405]: @nikitosing let's move transaction position display to a separate PR and make UI similar to Etherscan

**Review by vbaranov — COMMENTED** [review]
@nikitosing almost ready for merging. Please rebase onto remote master branch.

**Comment by nikitosing** [comment]
@vbaranov rebased

**Review by vbaranov — APPROVED** [review]
